src/openforms/registrations/contrib/zgw_apis/service.py

```diff
@@ -35,9 +36,14 @@ def create_zaak(options: dict) -> dict:
 
 
 def create_document(
-    name: str, submission_report: SubmissionReport, options: dict
+    name: str,
+    submission_report: SubmissionReport,
+    options: dict,
+    config: Optional[SingletonModel] = None,
 ) -> dict:
-    config = ZgwConfig.get_solo()
+    if not config:
+        config = ZgwConfig.get_solo()
```

Comment:
I don't like how this implicitly requires the model fields to have the same name. Leaving it in for now, but please pay attention to keeping a clean Python interface, it's already all complicated enough as it is.

Reply:
refactored it into a getter function